src/common.jl

```diff
@@ -30,3 +30,16 @@ struct Identity end
 \(::Identity, x) = copy(x)
 ldiv!(::Identity, x) = x
 ldiv!(y, ::Identity, x) = copyto!(y, x)
+
+"""
+Conjugated and unconjugated dot products
+"""
+abstract type DotType end
```

Comment:
I would call this `AbstractDot` in keeping with the usual Julia naming conventions.